Update Service Catalog Setup to include Backstage YAML support (#18708)
DataDog/documentation: content/en/tracing/service_catalog/setup.md

This GitHub Action allows you to register your services with the Service Catalog using a GitHub Action, with full control over when this information is sent to Datadog, and implement other compliance checks unique to your organization.

> ## Import Data from Backstage 

Reviewer: ```suggestion
## Import data from Backstage 
```

If you already have data or services registered in Backstage, you can import these services into Datadog directly. 

> If you use API / Terraform, replace the content of YAML content in your requests. 

Reviewer: ```suggestion
If you use API or Terraform, replace the content of YAML content in your requests. 
```

> If you use GitHub integration, directly save your Backstage YAMLs to a repo with Datadog read permission. Note that the file must be located at the root folder of a repo, and the file must be named `service.datadog.yaml`. 

Reviewer: ```suggestion
If you use GitHub integration, directly save your Backstage YAMLs to a repo with Datadog read permission.

**Note**: The file must be named `service.datadog.yaml` and be located at the root folder of a repo.
```

> It’s a partial translation, specifically, 

Reviewer: What is a partial translation? Please make this a complete sentence :)

Author: How about 'Datadog only recognize selected fields in Backstage YAMLs, specifically,

> - We only recognize `kind:component` and `spec.type=service` in Backstage YAMLs as services in Datadog

Reviewer: ```suggestion
- Datadog only recognizes `kind:component` and `spec.type=service` in Backstage YAMLs as services
```

> - `name` will be converted to `DD-SERVICE`

Reviewer: ```suggestion
- `name` gets converted to `DD-SERVICE`
```

> - `namespace` values will be mapped to custom tags

Reviewer: ```suggestion
- `namespace` values get mapped to custom tags
```

> - `lifecycle` will be mapped to `lifecycle`

Reviewer: ```suggestion
- `lifecycle` gets mapped to `lifecycle`
```

> - `owner` will be mapped to `team`

Reviewer: ```suggestion
- `owner` gets mapped to `team`
```

> - `metadata.links` will be mapped to `links`

Reviewer: ```suggestion
- `metadata.links` gets mapped to `links`
```

> - `metadata.description` will be mapped to `description`

Reviewer: ```suggestion
- `metadata.description` gets mapped to `description`
```

> - Other `specs` value will be mapped to custom tags

Reviewer: ```suggestion
- Other `specs` values get mapped to custom tags
```

This GitHub Action allows you to register your services with the Service Catalog using a GitHub Action, with full control over when this information is sent to Datadog, and implement other compliance checks unique to your organization.

## Import data from Backstage 

If you already have data or services registered in Backstage, you can import these services into Datadog directly. 

If you use API or Terraform, replace the content of YAML content in your requests. 
If you use GitHub integration, directly save your Backstage YAMLs to a repo with Datadog read permission.

**Note**: The file must be named `service.datadog.yaml` and be located at the root folder of a repo.

> It’s a partial translation, specifically, 

Reviewer: ```suggestion
Datadog only recognizes selected fields in Backstage YAML. Specifically: 
```
